src/credentials/loginManager.ts

```diff
@@ -55,4 +69,45 @@ export class LoginManager {
     public async logout(): Promise<void> {
         await this.awsContext.setCredentials(undefined)
     }
+
+    /**
+     * Updates the CredentialsStore if the credentials are considered different
+     */
+    private async updateCredentialsStore(credentialsId: string, provider: CredentialsProvider): Promise<void> {
+        const currentHash = this.credentialsStore.getCredentialsHashCode(credentialsId)
+        if (provider.getHashCode() !== currentHash) {
+            getLogger().verbose(`Credentials for ${credentialsId} have changed, using updated credentials.`)
+            this.credentialsStore.invalidateCredentials(credentialsId)
+        }
+
+        await this.credentialsStore.getCredentialsOrCreate(credentialsId, async () => {
```

Comment:
is `getCredentialsOrCreate` used anywhere else other than tests? It doesn't seem like it needs to take a function, since this create is generic enough to always be used

Reply:
This is the only place it is used. It was written in advance of this PR, and now that we have the actual use case formed in this PR, I could see passing the provider instead of a function. Thoughts?